### PR TITLE
Add documentation how to use allure reporting

### PR DESCRIPTION
**Pull request by xtrey**
Add documentation how to install and basic usage example of the allure reporting tool. Fix typo test/README.md

Related: https://github.com/scylladb/qa-tasks/issues/1665

Depends on: https://github.com/scylladb/scylladb/pull/18169

### CODE REVIEW

**Comment by temichus** [comment]
blocked by https://github.com/scylladb/scylladb/pull/18169

File: docs/dev/testing.md
Comment (temichus) [inline docs/dev/testing.md:303]: please  remove unnecessary changes

File: test/README.md
Comment (temichus) [inline test/README.md:17]: please  remove unnecessary changes

File: test/README.md
Comment (temichus) [inline test/README.md:24]: please  remove unnecessary changes

**Review by temichus — APPROVED** [review]

File: docs/dev/testing.md
Comment (piodul) [reply docs/dev/testing.md:326]: This is both an ask for help and a suggestion to put more detailed information here.

Comment (kbr-scylla) [reply docs/dev/testing.md:326]: Actually, I don't think we should document how to install Allure here at all. Why? They provide their documentation and keep it up-to-date: https://allurereport.org/docs/install-for-linux/ we should just link to it.

Same with the "manual allure installation" section below.

You should use the officially documented method of installation.

File: docs/dev/testing.md
Comment (kbr-scylla) [inline docs/dev/testing.md:352]: This is ambiguous. Is this `testlog`? Or do I need to go deeper into a specific mode or test run?

**Review by kbr-scylla — CHANGES_REQUESTED** [review]

File: docs/dev/testing.md
Comment (xtrey) [reply docs/dev/testing.md:352]: Added more information to make it less ambiguous.

File: docs/dev/testing.md
Comment (piodul) [reply docs/dev/testing.md]: Yes, it's much more clear now, thanks. However, I agree with Kamil that it's probably better to link to the Allure documentation right away as all this information is there and they should keep it up to date.

File: docs/dev/testing.md
Comment (piodul) [inline docs/dev/testing.md:308]: Nit: `module allure-pytest module` - "module" unnecessarily appears twice.

Comment (piodul) [reply docs/dev/testing.md:308]: "...and additional parameter added..." -> "...and __an__ additional parameter __is__ added..."

File: docs/dev/testing.md
Comment (piodul) [inline docs/dev/testing.md:342]: Kamil suggested dropping this part and I agree. Instructions on how to install allure, manually or not, are present on their website - and I suppose that they are maintaining those instructions. If you keep these instructions then we need to keep them in sync in case something changes there - it's an additional effort which I'm not sure is needed.

You could just provide a link to their documentation here instead - ofc it's also prone to breaking, but it's less effort maintaining just the link and easier to work around when it's not working.

I won't insist too strongly, but I think that dropping it would really be beneficial.

File: docs/dev/testing.md
Comment (piodul) [inline docs/dev/testing.md:353]: Nit: `please refer` -> `please refer to`

**Review by piodul — COMMENTED** [review]
Left some nitpicks, otherwise looks good. I managed to install the allure tool and display a report from the tests that I ran.

File: docs/dev/testing.md
Comment (piodul) [inline docs/dev/testing.md:323]: Nit: the backslash (`\`) looks unintentional and appears in the rendered output (see https://github.com/xtrey/scylladb/blob/allure_documentation/docs/dev/testing.md#basic-allure-usage). Perhaps you meant `:`?

**Review by piodul — APPROVED** [review]
One nitpick, looks good otherwise. Pre-approving, I'll merge after the nitpick is fixed and we get a green CI.

File: docs/dev/testing.md
Comment (xtrey) [reply docs/dev/testing.md]: Indeed, this is unintentional. Removed.

**Review by kbr-scylla — APPROVED** [review]

**Comment by piodul** [comment]
Queued. Sorry for the delay, I thought that we would have to wait for a green CI run, but apparently it is smart enough not to run for internal (developer) documentation changes.

**Pull request closed without merging**
